File: backend/GymRecommender/models/gym.py
from django.db import models
from .currency import Currency
from .account import Account
from .city import City
import uuid

class Gym(models.Model):
    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
    latitude = models.FloatField()
    longitude = models.FloatField()
    name = models.CharField(max_length=80)
    external_place_id = models.CharField(unique=True, max_length=50)
    external_rating = models.DecimalField(max_digits=4, decimal_places=2)
    external_rating_number = models.IntegerField()
    phone_number = models.CharField(max_length=15, blank=True, null=True)
    address = models.TextField()
    website = models.CharField(max_length=255, blank=True, null=True)
    is_wheelchair_accessible = models.BooleanField()
    monthly_mprice = models.DecimalField(max_digits=5, decimal_places=2, blank=True, null=True)
    yearly_mprice = models.DecimalField(max_digits=5, decimal_places=2, blank=True, null=True)
    six_months_mprice = models.DecimalField(max_digits=5, decimal_places=2, blank=True, null=True)
    created_at = models.DateTimeField()
    price_changed_at = models.DateTimeField(blank=True, null=True)
    changed_at = models.DateTimeField(blank=True, null=True)
    internal_rating = models.DecimalField(max_digits=4, decimal_places=2)
    internal_rating_number = models.IntegerField()
    congestion_rating = models.DecimalField(max_digits=4, decimal_places=2)
    congestion_rating_number = models.IntegerField()
    owned_by = models.ForeignKey(Account, models.DO_NOTHING, db_column='owned_by', blank=True, null=True)
    currency = models.ForeignKey(Currency, models.DO_NOTHING)
    city = models.ForeignKey(City, models.DO_NOTHING)

    class Meta:
        managed = False
        db_table = 'gym'

File: backend/Models/Gym.cs
using System.Text.Json.Serialization;
using backend.Enums;

namespace backend.Models;

public partial class Gym
{
    public Guid Id { get; set; }

    public double Latitude { get; set; }

    public double Longitude { get; set; }

    public string Name { get; set; } = null!;

    public string ExternalPlaceId { get; set; } = null!;

    public decimal ExternalRating { get; set; }

    public int ExternalRatingNumber { get; set; }

    public string? PhoneNumber { get; set; }

    public string Address { get; set; } = null!;

    public string? Website { get; set; }

    public bool IsWheelchairAccessible { get; set; }

    public decimal? MonthlyMprice { get; set; }

    public decimal? YearlyMprice { get; set; }

    public decimal? SixMonthsMprice { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? PriceChangedAt { get; set; }

    public DateTime? ChangedAt { get; set; }

    public decimal InternalRating { get; set; }

    public int InternalRatingNumber { get; set; }

    public decimal CongestionRating { get; set; }

    public int CongestionRatingNumber { get; set; }

    public Guid? OwnedBy { get; set; }

    public Guid CurrencyId { get; set; }

    public Guid CityId { get; set; }

    public virtual ICollection<Bookmark> Bookmarks { get; set; } = new List<Bookmark>();

    public virtual City City { get; set; } = null!;

    public virtual ICollection<CongestionRating> CongestionRatings { get; set; } = new List<CongestionRating>();

    public virtual Currency Currency { get; set; } = null!;

    public virtual ICollection<GymWorkingHour> GymWorkingHours { get; set; } = new List<GymWorkingHour>();
    [JsonIgnore] public virtual Account? OwnedByNavigation { get; set; }
    [JsonIgnore] public virtual ICollection<Ownership> Ownerships { get; set; } = new List<Ownership>();
    [JsonIgnore] public virtual ICollection<Rating> Ratings { get; set; } = new List<Rating>();
    [JsonIgnore] public virtual ICollection<Recommendation> Recommendations { get; set; } = new List<Recommendation>();

    public decimal? GetPrice(MembershipLength membershipLength)
    {
        return membershipLength switch
        {
            MembershipLength.Month => MonthlyMprice,
            MembershipLength.HalfYear => SixMonthsMprice,
            MembershipLength.Year => YearlyMprice,
            _ => MonthlyMprice
        };
    }
}